### PR TITLE
Zig bindings for orca (still WIP)

### PR DESCRIPTION
**Pull request by martinfouilleul**
_Imported from gitea. See https://git.handmade.network/hmn/orca/pulls/140 for previous conversation_

This is an incomplete set of zig bindings for orca. The included zig sample shows how to build and link the orca runtime with the zig app. It also shows how to use the orca bindings and provides a decent point of comparison to some of the C samples. It also proves that it works. ;)
Some of the API bindings are only partially complete, such as the oc_str8/16/32 functions, and others I haven't started at all, such as UI and GLES.
I tried to keep to zig paradigms and naming conventions, so in some cases functions are partially or entirely reimplemented in zig (such as a lot of the string code). Definitely welcome any feedback! And if we feel this PR isn't ready yet, we can wait for the bindings to be done before merging.

### CODE REVIEW

File: src/orca.zig [hunk 1]
```diff
@@ -0,0 +1,3822 @@
...
+    extern fn oc_mat2x3_inv(x: Mat2x3) Mat2x3;
+    extern fn oc_mat2x3_rotate(radians: f32) Mat2x3;
+    extern fn oc_mat2x3_translate(x: f32, y: f32) Mat2x3;
+
+    extern fn oc_matrix_push(matrix: Mat2x3) void;
+    extern fn oc_matrix_pop() void;
+    extern fn oc_matrix_top() Mat2x3;
+
+    pub const mul = oc_mat2x3_mul;
+    pub const mul_m = oc_mat2x3_mul_m;
```
Comment (ilidemi) [inline src/orca.zig:1685]: nit: `mulM`

**Comment by ilidemi** [comment]
There are a few APIs left with Str8/Str32: `utf8`, `Canvas.glyphOutlines` and neighbors, `openWithDialog`. Guess there are a few in `InputState` as well. Do we want the APIs to be clean-clean or stick with good enough?

**Comment by rdunnington** [comment]
> There are a few APIs left with Str8/Str32: `utf8`, `Canvas.glyphOutlines` and neighbors, `openWithDialog`. Guess there are a few in `InputState` as well. Do we want the APIs to be clean-clean or stick with good enough?

Yeah I think I should do a pass on those, I thought I had already but obviously I missed some so I'll look again. Thanks!

**Comment by rdunnington** [comment]
These bindings have moved to their own repo here: https://github.com/orca-app/orca-zig . A couple of PRs affecting the C source have been extracted: 
* #40 
* #41 
Closing this PR since it's no longer relevant.

**Pull request closed without merging**